### ISSUE TITLE
Signing mismatch between web3 and Java encoding

### ISSUE THREAD

**Issue by qoire** [aionnetwork/aion_rlp.js#1]

**Comment by arajasek** [comment]
Web3.eth.accounts.signTransaction() is incorrectly encoding the "value" field in some cases.

Sample code: 

```
var transferBal = '203261307462270040000000000';

var tx = { gas: 2000000, gasPrice: 10000000000, value: transferBal, from: addr1, to: addr2 };

var signedTx = web3.eth.accounts.signTransaction(tx, richKey);

var rsp = signedTx.then( (sTx) => web3.eth.sendSignedTransaction(sTx.rawTransaction)
	.on('transactionHash', txHash => { console.log("txHash", txHash) })
    .on('receipt', receipt => { console.log("receipt", receipt) }))
	.catch( (err) => console.log("ERROR", err) );
```

This snippet fails, because the kernel complains that the value field is 
```
value=20646261356015024812400375159853539737589118092617902863112220720
```

When the value field of the transaction is 123456789, the **actual value transferred** is 907507751940624169017. This succeeded because I happened to be using a test account with a sufficiently large balance.

The privateKey, `richKey`, was `0x75cbf70fe05ac094f3d7ac1c2d78e086901227f9b6d9bd42fc43fff7b04bdab4d12f98477dc1d5eebc1d3d7bd5743625a03b4992dd6be7612c9531d821530000`

The sender address, `addr1`, was `0xa0bf1c0cbfda0c829fe3520f4a0803c060b2d5bff2381f0c0b8ae4bf27379166`

The recipient address, `addr2`, was 
`0xA0e855B1173E17D87Bb63e33fdFF36E6494fD2CAC42bc7f80d0a093Ee29d263A`

**Comment by aionick** [comment]
@arajasek and I have confirmed that this issue **does** exist and it is not a simple mix-up between hexadecimal and decimal conversions. We suspected that the value we thought we were entering in decimal was actually being interpreted as hex - this is **wrong** and is not what is happening.

The issue seems to reside specifically in the command `web3.eth.accounts.signTransaction()`. After calling this function we then took the raw encoded bytes and decoded them using the `AionTransaction` class's rlp decoding and this produced a transaction with the incorrect value. 

This means that `web3.eth.sendSignedTransaction()` is most likely correct and is not the source of the issue.

We should probably be patching this and verifying _all_ of the fields of the transaction are correct, not just value.
